lib/utilities/strings.rb

```diff
@@ -96,12 +97,17 @@ def self.authorship_sentence(last_names = [])
     last_names.to_sentence(two_words_connector: ' & ', last_word_connector: ' & ')
   end
 
+  # Splits a string on special characters, returning an array of the strings that do not contain digits.
+  #
+  # It splits on accent characters, and does not split on underscores. The method is used for building wildcard searches,
+  # so splitting on accents creates pseudo accent insensitivity in searches.
+  #
   # @param string [String]
   # @return [Array]
-  #   whitespace split, then any string containing a digit eliminated
+  #   whitespace and special character split, then any string containing a digit eliminated
   def self.alphabetic_strings(string)
     return [] if string.nil? || string.length == 0
-    string.split(/\W/).select{|b| !(b =~ /\d/) }
+    string.split(/\W/).select{|b| !(b =~ /\d/) }.reject { |b| b.empty? }
```

Comment:
Would `string&.split(/\W/)&.reject{ |b| b.blank? || b =~ /\d/ } || []` work as replacement for the entire code of this method?

Reply:
IMO it's a lot harder to read as a one-liner, this is clear how it works.